### DISCUSSION TITLE
Issue with using custom user_basis for a constrained Hilbert space

### DISCUSSION THREAD

**Discussion by tanmaybhore** [QuSpin/QuSpin#695] [Q&A]
I've been trying to implement the example of a custom user-basis, given in the example here: [Quantum scar Hamiltonian](https://quspin.github.io/QuSpin/examples/example14.html?)

However, upon running the code given above, I got the error that np.int is deprecated in the current version of numpy and can be replaced by simply int. I changed all instances of np.int to int in the user_basis source code, however, now I get the error when I try to print the basis: 

_AttributeError: module 'numpy' has no attribute 'object'.
`np.object` was a deprecated alias for the builtin `object`. To avoid this error in existing code, use `object` by itself. Doing this will not modify any behavior and is safe._ 

I have not had this issue with using the standard spin_basis_general class. This is only coming up when I use the custom defined user_basis.

I have the latest version of quspin (0.3.7) and I'm running Python 3.11.7. Is this an issue with what version of numpy I have? Or is something wrong with my quspin installation? Any help is appreciated!

**Comment by ToastyTomatos** [comment]
Try downgrading to an older version of numpy, f.e. 1.22.0
